extension/src/minn_maxn.rs

```diff
@@ -0,0 +1,1177 @@
+use pgx::{iter::SetOfIterator, *};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    aggregate_utils::in_aggregate_context,
+    flatten,
+    palloc::{Inner, Internal, InternalAsValue, ToInternal},
+    pg_type,
+    raw::bytea,
+    ron_inout_funcs,
+};
+
+use ordered_float::NotNan;
+
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+use std::marker::PhantomData;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct NMostTransState<T: Ord> {
+    capacity: usize,
+    heap: BinaryHeap<T>,
+    phantom: PhantomData<T>,
```

Comment:
The `PhantomData` is unnecessary, since the type parameter is used by the `BinaryHeap<T>` (which indirectly [uses a `PhantomData` internally](https://github.com/rust-lang/rust/blob/4ca19e09d302a4cbde14f9cb1bc109179dc824cd/library/core/src/ptr/unique.rs#L42)).

Reply:
Another good catch.